src/main/java/uk/ac/ebi/ena/readtools/utils/Utils.java

```diff
@@ -42,8 +44,7 @@ public static void replaceUracilBasesInFastq(String inputFastq, String outputFas
         for (FastqRecord record : reader) {
             String result = record.getReadString();
 
-            Matcher matcher = URACIL_PATTERN.matcher(result);
-            if (matcher.find()) {
+            if (matcher.reset(result).find()) {

```

Comment:
Ok, I understand that the reset is used instead of creating a new Matcher to optimize code execution.